src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx

```diff
@@ -0,0 +1,279 @@
+
+In this section, you will learn how to integrate Amazon Rekognition into your application using AWS Amplify, leveraging its powerful image analysis capabilities seamlessly.
+
+## Step 1 - Set up the project
+
+Set up your project by following the instructions in the [Quickstart guide](/[platform]/start/quickstart/). 
+
+## Step 2 - Install Rekognition Libraries
+We'll create a new API endpoint that'll use the the AWS SDK to call the Amazon Rekognition service. To install the Amazon Rekognition SDK, run the following command in your project's root folder:
+
+```bash
+npm install @aws-sdk/client-rekognition
+
+```
+
+## Step 3 - Setup Storage
+
+Create a file named `amplify/storage/resource.ts` and add the following content to configure a storage resource:
+
+
+```ts title="amplify/storage/resource.ts"
+import { defineStorage } from '@aws-amplify/backend';
+
+export const storage = defineStorage(
+  name: 'predictions_gen2'
+});
+
+```
+
+## Step 4 - Add your Amazon Rekognition as Datasource
+
+To use the Amazon Rekognition service, you need to add Amazon Rekognition as an HTTP Data Source and configure the proper IAM policy for Lambda to effectively utilize the desired feature and grant permission to access the storage. In this case, you can add the `rekognition:DetectText` and `rekognition:DetectLabels` actions to the policy. Update the `amplify/backend.ts` file as shown below.
+
+ ```ts title= "amplify/backend.ts"
+
+import { defineBackend } from '@aws-amplify/backend';
+import { auth } from './auth/resource';
+import { data } from './data/resource';
+import { Stack } from 'aws-cdk-lib';
+import { PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { storage } from './storage/resource';
+
+const backend = defineBackend({
+  auth,
+  data,
+  storage
+});
+
+const dataStack = Stack.of(backend.data)
+
+const rekognitionDataSource = backend.data.addHttpDataSource(
+  "RekognitionDataSource",
+  `https://rekognition.${dataStack.region}.amazonaws.com`,
+  {
+    authorizationConfig: {
+      signingRegion: dataStack.region,
+      signingServiceName: "rekognition",
+    },
+  }
+);
+
+rekognitionDataSource.grantPrincipal.addToPrincipalPolicy(
+  new PolicyStatement({
+    actions: ["rekognition:DetectText", "rekognition:DetectLabels"],
+    resources: ["*"],
+  })
+);
+
+backend.storage.resources.bucket.grantReadWrite(
+  rekognitionDataSource.grantPrincipal
+);
+
+```
+## Step 5 - Configure the function handler 
+
+Define the function handler by creating a new file, `amplify/data/identifyText.ts`. This function analyzes the image and extracts text using the Amazon Rekognition DetectText service.
+
+```ts title="amplify/data/identifyText.ts"
+
+export function request(ctx) {
+  return {
+    method: "POST",
+    resourcePath: "/",
+    params: {
+      body: {
+        Image: {
+          S3Object: {
+            Bucket:
+              "<replace with S3 Bucket name>",
```

Comment:
Is there a way to set an AppSync environment variable here instead?

Reply:
updated code using env variable.